src/util.ts

```
@@ -29,11 +29,18 @@ export async function writeFile(fileContent: string): Promise<string> {
   if (!workspace) {
     throw new Error('Missing GITHUB_WORKSPACE!');
   }
-  const kubeConfigPath = path.join(workspace, uuidv4());
+
+  // Generate a random filename to store the credential. 12 bytes is 24
+  // characters in hex. It's not the ideal entropy, but we have to be under the
+  // 255 character limit for Windows filenames (which includes their entire
+  // leading path).
+  const uniqueName = randomBytes(12).toString('hex');
+  const kubeConfigPath = pathjoin(workspace, uniqueName);
+
   try {
-    await fs.writeFileSync(kubeConfigPath, fileContent);
+    await fs.writeFile(kubeConfigPath, fileContent);
     return kubeConfigPath;
   } catch (err) {
-    throw new Error(`Unable to write kubeconfig to file: ${kubeConfigPath}`);
+    throw new Error(`Failed to write kubeconfig to ${kubeConfigPath}: ${err}`);
```

Twas bug: we were swallowing the root cause of the error